src/common/databaseFilters.test.ts

```diff
@@ -0,0 +1,426 @@
+import { beforeEach, expect, it } from "@jest/globals";
+import { familySearch, fullNameSearch, phoneSearch, postcodeSearch } from "./databaseFilters";
+import { DbParcelRow } from "@/databaseUtils";
+import { DbQuery, ServerSideFilterMethod } from "@/components/Tables/Filters";
+
+const logID = "a2adb0ba-873e-506b-abd1-8cd1782923c8";
+
+jest.mock("@/logger/logger", () => ({
+    logErrorReturnLogId: jest.fn(() => Promise.resolve(logID)),
+}));
+
+jest.mock("@/supabaseClient", () => {
+    return { default: jest.fn() };
+});
+
+const mockQueryWithOrAppended = jest.fn();
+const mockOr: jest.Mock = jest.fn(() => mockQueryWithOrAppended);
+
+describe("test for full name filter", () => {
```

Comment:
There were enough special cases for each filter that it seemed to be worth having all the tests in this file.